File: src/main/Frontend/myApp/src/app/class/Animal.ts
import {Shelter} from './Shelter';
import {AnimalType} from './AnimalType';
import {IsAdopted} from "./IsAdopted";

export class Animal {



  id = 0;
  name = '';
  type: AnimalType = new AnimalType();
  virtual_owner: IsAdopted = new IsAdopted();
  breed = '';
  gender = 0;
  age = 0;
  lifestory = '';
  imagine = '';
  illnes = '';
  gonadectomy = 0;
  size = 0;
  shelter: Shelter = new Shelter();
  isadopted: IsAdopted = new IsAdopted();
  owner = '';
  thestatus=0;






}

File: src/main/java/com/example/demo/dal/entities/Animal.java
package com.example.demo.dal.entities;

import lombok.*;

import javax.persistence.*;
import java.util.Optional;

@Entity
@Data
@AllArgsConstructor // minden argumentummla konstruktor
@NoArgsConstructor // ures konstruktor
@EqualsAndHashCode(callSuper = true) // legenerálja
@Table(name = "animals") // tábla elnevezés
public class Animal extends BaseEntity {

    private String name;

    private int animaltype_id;

    private String virtual_owner; // int

    private String breed;

    private int gender;

    private int age;

    private String illnes;

    private String imagine;

    private int gonadectomy;

    private int size;

    @Lob
    private String lifestory;

    private int image;

    private int shelter_id;

}
